source/common/network/utility.h

```diff
@@ -155,6 +162,14 @@ class Utility {
                                                             uint32_t port);
 
   /**
+   * Returns an any address given the remote IP address. Returns the IPv4 any address by default.
+   * @param address the remote IP address.
+   * @returns the IPv6 any address if the remote address is in the IPv6 family. Otherwise, returns
+   * the IPv4 any address.
+   */
+  static Address::InstanceConstSharedPtr getNullLocalAddress(const Address::Instance& address);
```

Comment:
This is a bit weird. The null local address is really a hack to avoid populating the local address in `ConnectionImpl`. I don't think we want a general purpose utility function for this. You could put it in the anonymous namespace in `connection_impl.cc`.